src/validation/json-schemas/public-jwk.json

```
@@ -0,0 +1,68 @@
+{
+  "$id": "https://identity.foundation/dwn/json-schemas/public-jwk.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "required": ["kty", "crv", "x"],
+  "properties": {
+    "alg": {
+      "type": "string"
+    },
+    "kid": {
+      "type": "string"
+    },
+    "kty": {
+      "enum": ["EC", "RSA", "oct", "OKP"]
+    },
+    "crv": {
+      "type": "string"
+    },
+    "x": {
+      "type": "string"
+    },
+    "y": {
+      "type": "string"
+    }
+  },
+  "anyOf": [
+    {
+      "properties": {
+        "kty": {
+          "const": "OKP"
+        },
+        "crv": {
+          "enum": ["Ed25519"]
+        }
+      }
+    },
+    {
+      "properties": {
+        "kty": {
+          "const": "EC"
+        },
+        "crv": {
+          "enum": ["secp256k1"]
+        }
+      }
+    }
+  ],
+  "not": {
+    "anyOf": [
+      {
+        "properties": {
+          "kty": {
+            "const": "EC"
+          }
+        },
+        "anyOf": [{ "required": ["d"] }]
+      },
+      {
+        "properties": {
+          "kty": {
+            "const": "OKP"
+          }
+        },
+        "anyOf": [{ "required": ["d"] }]
+      }
+    ]
+  }
```

Would the below be sufficient?
```suggestion
"additionalProperties": false
```

---

@thehenrytsai  the reason I did the above is 3-fold, but could be offbase so let me know your thoughts. 
1. If and  when new curves/keytypes are supported that don't have "d" as a private property
2. There are a lot of different optional properties that can exist in the publicJwk so didn't want to just restrict everything
3. This might allow us to abstract away some of this validation into a common jwk schema that includes private and public properties and then use the "not" above to validate that it's public.

---

You definitely have given it way more thought that I do! Sounds logical!